yb_migrate/cmd/importDataFileCommand.go

```diff
@@ -0,0 +1,264 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/yugabyte/yb-db-migration/yb_migrate/src/datafile"
+	"github.com/yugabyte/yb-db-migration/yb_migrate/src/tgtdb"
+	"github.com/yugabyte/yb-db-migration/yb_migrate/src/utils"
+)
+
+var (
+	fileType            string
+	delimiter           string
+	dataDir             string
+	fileTableMapping    string
+	hasHeader           bool
+	tableNameVsFilePath = make(map[string]string)
+	supportedFileTypes  = []string{datafile.CSV, datafile.SQL}
+	importDataFileMode  bool
+)
+
+var importDataFileCmd = &cobra.Command{
+	Use:   "file",
+	Short: "This command imports data from given files into YugabyteDB database",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		importDataFileMode = true
+		checkFileType()
+		checkDataDirFlag()
+		parseFileTableMapping()
+		prepareForImportDataCmd()
+		importDataFile()
+	},
+}
+
+func importDataFile() {
+	utils.PrintAndLog("import of data in %q database started", target.DBName)
+	err := target.DB().Connect()
+	if err != nil {
+		utils.ErrExit("Failed to connect to the target DB: %s", err)
+	}
+	fmt.Printf("Target YugabyteDB version: %s\n", target.DB().GetVersion())
+
+	sourceDBType = ORACLE // dummy value
+
+	dataFileDescriptor = datafile.OpenDescriptor(exportDir)
+	targets := getYBServers()
+
+	var parallelism = parallelImportJobs
+	if parallelism == -1 {
+		parallelism = len(targets)
+	}
+	log.Infof("parallelism=%v", parallelism)
+
+	if loadBalancerUsed {
+		clone := target.Clone()
+		clone.Uri = getTargetConnectionUri(clone)
+		targets = []*tgtdb.Target{clone}
+	}
+	if target.VerboseMode {
+		fmt.Printf("Number of parallel imports jobs at a time: %d\n", parallelism)
+	}
+
+	if parallelism > SPLIT_FILE_CHANNEL_SIZE {
+		splitFileChannelSize = parallelism + 1
+	}
+	splitFilesChannel := make(chan *SplitFileImportTask, splitFileChannelSize)
+	targetServerChannel := make(chan *tgtdb.Target, 1)
+
+	go roundRobinTargets(targets, targetServerChannel)
+	generateSmallerSplits(splitFilesChannel)
+	go doImport(splitFilesChannel, parallelism, targetServerChannel)
+	checkForDone()
+
+	time.Sleep(time.Second * 2)
+	fmt.Printf("\nexiting...\n")
+}
+
+func prepareForImportDataCmd() {
+	CreateMigrationProjectIfNotExists("postgresql", exportDir)
+	createExportDataDoneFlag()
+	createDataFileSymLinks()
+	prepareCopyCommands()
+	setImportTableListFlag()
+
+	tableFileSize := getFileSizeInfo()
+	dfd := &datafile.Descriptor{
+		FileType:      fileType,
+		TableRowCount: tableFileSize,
```

Comment:
Can we rename the `TableRowCount` as `TotalProgressUnitCount`?

Can we always operate the progress-bars based on amount of input file read? This way we don't have to refer to row-count in one location and byte-count in other.

Reply:
@amit-yb  I had the same thought but the reason i feel to not move away from row count is that later on when we implement `validate` command, rc can be useful there. Also rightnow for export data pb we anyways count rows.

I really like your suggestion to store the progress information as file attribute this way we will able to retain the row counts info for each split in this new command also.